tools/clang/lib/SPIRV/AstTypeProbe.cpp

```diff
@@ -742,6 +742,21 @@ bool isRowMajorMatrix(const SpirvCodeGenOptions &spvOptions, QualType type) {
   return !spvOptions.defaultRowMajor;
 }
 
+bool containsMatrixMajorAttr(const ASTContext &astContext, QualType type,
+                             llvm::Optional<bool> *isRowMajorAttr) {
+  bool attrRowMajor = false;
+  if (hlsl::HasHLSLMatOrientation(type, &attrRowMajor)) {
+    *isRowMajorAttr = attrRowMajor;
```

Comment:
SPIR-V row-major is HLSL col-major and SPIR-V col-major is HLSL row-major.
So, this logic should be flipped. For example:
https://github.com/microsoft/DirectXShaderCompiler/blob/f4965b71dd878c42c219ec00d31c406c483fd045/tools/clang/lib/SPIRV/AstTypeProbe.cpp#L727-L729

Reply:
In general, I think you can remove this function, and just update the `isRowMajorMatrix` function in AstTypeProbe.cpp

`isRowMajorMatrix` is just missing the array of matrices case I think.